polybrain.el

```diff
@@ -41,92 +41,88 @@ changed."
+  "Get text content of the visualized entry."
   (save-excursion
-    (beginning-of-buffer)
-    (search-forward "--- Entry")
+    (goto-char org-brain--vis-entry-text-marker)
     (end-of-line)
-    (forward-char)
-    (let ((beg (point)))
-      (end-of-buffer)
-      (buffer-substring beg (point)))))
+    ;; (forward-char)
+    (buffer-substring (point) (point-max))))
 
 (defun polybrain--save-contents (content)
+  "Replace the entry's text with CONTENT."
   (find-file (org-brain-entry-path org-brain--vis-entry))
-  (beginning-of-buffer)
-  (goto-char
-   (or (save-excursion
-         (while (and (re-search-forward org-brain-keyword-regex nil t) (org-before-first-heading-p))
-           (end-of-line)
-           (forward-char))
-         (point))
-       (point-min)))
-  (insert content)
-  (delete-region (point) (point-max))
-  (save-buffer)
-  (switch-to-buffer (other-buffer (current-buffer) 1)))
+  (seq-let (entry-min entry-max) (org-brain-text-positions org-brain--vis-entry)
+    (goto-char entry-min)
+    (delete-region entry-min entry-max)
+    (insert content)
+    (unless (looking-at-p "\n")
+      (insert "\n\n"))
+    (save-buffer)
+    (switch-to-buffer (other-buffer (current-buffer) 1))))
 
 (defun polybrain-save ()
+  "Save entry text content to the entry's file."
   (interactive)
-  (org-brain-set-title org-brain--vis-entry org-brain--vis-entry)
-  (let ((content (polybrain--entry-contents)))
-    (polybrain--save-contents content)))
+  (polybrain--save-contents (polybrain--entry-contents)))
 
 (defvar polybrain--last-brain-point nil)
 (defvar polybrain--last-org-point nil)
 
 (defun polybrain-switch ()
-    (interactive)
-    (when polybrain-save-on-switch
-      (polybrain-save))
-    (seq-let (org-min brain-max) (polybrain--get-point-min-max)
-      (let ((p (point)))
-        (if (search-forward "--- Entry" nil t)
-            (progn
-              (setq polybrain--last-brain-point p)
-              (if polybrain--last-org-point
-                  (goto-char (max org-min polybrain--last-org-point))
-                (end-of-line) (forward-char)))
-          (setq polybrain--last-org-point p)
-          (if polybrain--last-brain-point
-              (goto-char (floor (min brain-max polybrain--last-brain-point)))
-            (beginning-of-buffer))))))
+  "Move `point' between `org-brain-visualize' and `org-mode' content."
+  (interactive)
+  (seq-let (org-min brain-max) (polybrain--get-point-min-max)
+    (let ((p (point)))
+      (if (< p org-brain--vis-entry-text-marker)
+          (progn
+            ;; In case the text has been changed outside org-brain-visualize
+            (revert-buffer)
+            (setq polybrain--last-brain-point p)
+            (if polybrain--last-org-point
+                (goto-char (max org-min polybrain--last-org-point))
+              (goto-char org-brain--vis-entry-text-marker)
+              (end-of-line) (forward-char)))
+        (when polybrain-save-on-switch (polybrain-save))
```

Comment:
I figured we only needed to save when switching from the org-mode portion of the buffer.